app/services/bookmarkable.rb

```diff
@@ -74,7 +72,46 @@ def perform_search_query(bookmarks, query, ts_query)
   #                          array _not_ an ActiveRecord::Relation.
   def perform_preload(bookmarks)
     ActiveRecord::Associations::Preloader
-      .new(records: Bookmark.select_type(bookmarks, model.to_s), associations: [bookmarkable: preload_associations])
+      .new(
+        records: Bookmark.select_type(bookmarks, bookmarkable.model.to_s),
+        associations: [bookmarkable: bookmarkable.preload_associations]
```

Comment:
When `bookmarkable.preload_associations` is an empty array or `nil`, I think we can just skip this whole preloader thing.  Perhaps `BaseBookmarkable` can implement a `has_preloads` method that is set on initialization?

Reply:
Oh yes you are right good catch :+1: